### ISSUE TITLE
Adjustments of already-adjusted standard errors

### ISSUE THREAD

**Issue by elbersb** [vincentarelbundock/modelsummary#280]
This one is maybe more tricky, and it may be impossible to deal with this correctly in all circumstances. But first the problem: I estimate a `fixest` model and cluster the standard errors. I then request iid standard errors in `modelsummary`, which doesn't work -- `modelsummary` just reports the already-clustered standard errors.

``` r
library("modelsummary")
library("fixest")
mod <- feols(hp ~ mpg + drat, mtcars, cluster = "vs")
modelsummary(mod, vcov = "iid", output = "markdown")
```

|             | Model 1  |
|:------------|:--------:|
| (Intercept) | 278.515  |
|             | (20.573) |
| mpg         |  -9.985  |
|             | (4.110)  |
| drat        |  19.126  |
|             | (32.644) |
| Std. Errors |   IID    |

``` r
coef(summary(lm(hp ~ mpg + drat, data = mtcars)))
#>               Estimate Std. Error    t value     Pr(>|t|)
#> (Intercept) 278.515455  55.414866  5.0260061 2.359726e-05
#> mpg          -9.985499   1.791837 -5.5727709 5.172030e-06
#> drat         19.125752  20.197756  0.9469246 3.515013e-01
```

<sup>Created on 2021-05-06 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.4 (2021-02-15)
#>  os       macOS Big Sur 10.16         
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       Europe/Brussels             
#>  date     2021-05-06                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package      * version    date       lib source        
#>  assertthat     0.2.1      2019-03-21 [1] CRAN (R 4.0.2)
#>  backports      1.2.1      2020-12-09 [1] CRAN (R 4.0.2)
#>  broom          0.7.6      2021-04-05 [1] CRAN (R 4.0.2)
#>  checkmate      2.0.0      2020-02-06 [1] CRAN (R 4.0.2)
#>  cli            2.5.0      2021-04-26 [1] CRAN (R 4.0.4)
#>  colorspace     2.0-0      2020-11-11 [1] CRAN (R 4.0.2)
#>  crayon         1.4.1      2021-02-08 [1] CRAN (R 4.0.2)
#>  DBI            1.1.1      2021-01-15 [1] CRAN (R 4.0.2)
#>  digest         0.6.27     2020-10-24 [1] CRAN (R 4.0.2)
#>  dplyr          1.0.5      2021-03-05 [1] CRAN (R 4.0.2)
#>  dreamerr       1.2.3      2020-12-05 [1] CRAN (R 4.0.2)
#>  ellipsis       0.3.2      2021-04-29 [1] CRAN (R 4.0.2)
#>  evaluate       0.14       2019-05-28 [1] CRAN (R 4.0.1)
#>  fansi          0.4.2      2021-01-15 [1] CRAN (R 4.0.2)
#>  fixest       * 0.8.4      2021-03-29 [1] CRAN (R 4.0.2)
#>  Formula        1.2-4      2020-10-16 [1] CRAN (R 4.0.2)
#>  fs             1.5.0      2020-07-31 [1] CRAN (R 4.0.2)
#>  generics       0.1.0      2020-10-31 [1] CRAN (R 4.0.2)
#>  glue           1.4.2      2020-08-27 [1] CRAN (R 4.0.2)
#>  highr          0.9        2021-04-16 [1] CRAN (R 4.0.2)
#>  htmltools      0.5.1.1    2021-01-22 [1] CRAN (R 4.0.2)
#>  httr           1.4.2      2020-07-20 [1] CRAN (R 4.0.2)
#>  kableExtra     1.3.4      2021-02-20 [1] CRAN (R 4.0.2)
#>  knitr          1.33       2021-04-24 [1] CRAN (R 4.0.2)
#>  lattice        0.20-41    2020-04-02 [1] CRAN (R 4.0.4)
#>  lifecycle      1.0.0      2021-02-15 [1] CRAN (R 4.0.2)
#>  magrittr       2.0.1      2020-11-17 [1] CRAN (R 4.0.2)
#>  modelsummary * 0.7.0.9000 2021-05-04 [1] local         
#>  munsell        0.5.0      2018-06-12 [1] CRAN (R 4.0.2)
#>  nlme           3.1-152    2021-02-04 [1] CRAN (R 4.0.4)
#>  numDeriv       2016.8-1.1 2019-06-06 [1] CRAN (R 4.0.2)
#>  pillar         1.6.0      2021-04-13 [1] CRAN (R 4.0.2)
#>  pkgconfig      2.0.3      2019-09-22 [1] CRAN (R 4.0.2)
#>  purrr          0.3.4      2020-04-17 [1] CRAN (R 4.0.2)
#>  R6             2.5.0      2020-10-28 [1] CRAN (R 4.0.2)
#>  Rcpp           1.0.6      2021-01-15 [1] CRAN (R 4.0.2)
#>  reprex         2.0.0      2021-04-02 [1] CRAN (R 4.0.2)
#>  rlang          0.4.11     2021-04-30 [1] CRAN (R 4.0.2)
#>  rmarkdown      2.7        2021-02-19 [1] CRAN (R 4.0.2)
#>  rstudioapi     0.13       2020-11-12 [1] CRAN (R 4.0.2)
#>  rvest          1.0.0      2021-03-09 [1] CRAN (R 4.0.2)
#>  sandwich       3.0-0      2020-10-02 [1] CRAN (R 4.0.2)
#>  scales         1.1.1      2020-05-11 [1] CRAN (R 4.0.2)
#>  sessioninfo    1.1.1      2018-11-05 [1] CRAN (R 4.0.2)
#>  stringi        1.5.3      2020-09-09 [1] CRAN (R 4.0.2)
#>  stringr        1.4.0      2019-02-10 [1] CRAN (R 4.0.2)
#>  styler         1.4.1      2021-03-30 [1] CRAN (R 4.0.2)
#>  svglite        2.0.0      2021-02-20 [1] CRAN (R 4.0.2)
#>  systemfonts    1.0.1      2021-02-09 [1] CRAN (R 4.0.2)
#>  tables         0.9.6      2020-09-22 [1] CRAN (R 4.0.2)
#>  tibble         3.1.1      2021-04-18 [1] CRAN (R 4.0.2)
#>  tidyr          1.1.3      2021-03-03 [1] CRAN (R 4.0.2)
#>  tidyselect     1.1.1      2021-04-30 [1] CRAN (R 4.0.2)
#>  utf8           1.2.1      2021-03-12 [1] CRAN (R 4.0.2)
#>  vctrs          0.3.8      2021-04-29 [1] CRAN (R 4.0.2)
#>  viridisLite    0.4.0      2021-04-13 [1] CRAN (R 4.0.2)
#>  webshot        0.5.2      2019-11-22 [1] CRAN (R 4.0.2)
#>  withr          2.4.2      2021-04-18 [1] CRAN (R 4.0.2)
#>  xfun           0.22       2021-03-11 [1] CRAN (R 4.0.2)
#>  xml2           1.3.2      2020-04-23 [1] CRAN (R 4.0.2)
#>  yaml           2.2.1      2020-02-01 [1] CRAN (R 4.0.2)
#>  zoo            1.8-9      2021-03-09 [1] CRAN (R 4.0.2)
#> 
#> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

</details>

I'm not yet sure what's the best way to deal with this--disallow adjustments of `fixest` models? Custom logic for `fixest` to deal with this? Would be interested in hearing your thoughts about this.

**Comment by vincentarelbundock** [comment]
Yeah, this one is a bit tricky. Maybe @grantmcdermott has better ideas for a user interface, but I think for now the best approach is to use a named list and the `fixest::vcov.fixest` method:

```r
library("modelsummary")
library("fixest")

mod <- feols(hp ~ mpg + drat, mtcars)

vc <- list(
  "Std" = vcov(mod),
  "Clust: vs" = vcov(mod, se = "cluster", cluster = "vs"))

modelsummary(mod, vcov = vc)
```
I have already hard-coded some `fixest` stuff because this is a package I use a lot in my own work, but if at all possible, I would much prefer to avoid going down the rabbit hole of having to support every nook and crannies of non-standard packages.

That said, I agree that it might be appropriate to raise a warning when users try to use `modelsummary`'s `vcov` argument with `fixest` models. And since I like `fixest` a lot, I'm not totally closed to the idea of including a couple more lines specific to `fixest`. But finding an easy way to "translate" the `vcov` into something that `fixest::vcov.fixest` will understand might be difficult...

**Comment by elbersb** [comment]
Okay, with a bit more reflection a few more thoughts. There seem to be two issues here: First, specifically about `fixest`, which I'm agnostic on. For maintainability it might be easier just to leave this out. There is always an option to supply a custom `vcov`, and both `modelsummary` and `fixest` make this very easy (as you show in your response). Second, a more general question about the use of the _convenient_ `vcov` options, such as "robust", "HC", "HC0", "HC1", "HC2", etc. 

My understanding is that all of these are implemented through `sandwich`. Shouldn't we be able to see whether the model supported by `sandwich`? If we encounter an object of class `fixest` or `lm_robust` (i.e., one unsupported by `sandwich`) **and** one of the 'convenient' options is supplied, `modelsummary` would error. This would be more of a whitelist than a blacklist, and it would automatically follow from `sandwich`. I don't think this proposal would restrict any useful functionality that is available now, but it would possibly avoid many misleading SE estimates. (Hope I'm not missing something obvious here...)

**Comment by vincentarelbundock** [comment]
Agreed on both points. 

Unless we (or some other contributor) can submit a PR with a very elegant `fixest` solution that takes few lines of code and is easy to maintain, let's not hardcode anymore.

A whitelist does sound like an improvement from the perspective of maintainability. But we would lose some things, wouldn't we? For example, `sandwich::vcovHC` does appear to work on `fixest` models, and I think the results are "correct". It seems desirable to keep the ability to use vcov labels like `stata` or `robust` with that kind of model. 

If you look at this code:

https://github.com/vincentarelbundock/modelsummary/blob/main/R/get_vcov.R#L17

You'll see that when we supply labels, we try `sandwich` and return a warning if it doesn't work. So in a sense, this is already a "whitelist" because it only works for models where `sandwich` works (assuming we go back to Error as decided).

The only real problem is related to this line here:

https://github.com/vincentarelbundock/modelsummary/blob/main/R/get_vcov.R#L20

When we return `NULL`, `modelsummary` will apply `vcov`, but that method can return different default values, not necessarily classical or iid.

**Comment by elbersb** [comment]
> But we would lose some things, wouldn't we? For example, `sandwich::vcovHC` does appear to work on `fixest` models, and I think the results are "correct". It seems desirable to keep the ability to use vcov labels like `stata` or `robust` with that kind of model.

Yes, that's true for `fixest` models that are estimated without fixed effects. But I think in practical applications (where probably most `feols` calls will involve at least one fixed effect - I may be wrong here), the result will be incorrect, because modelsummary will adjust a vcov matrix that has already been adjusted.

The problem is of course even more confusing, because `summary.fixest` does return the correct values. So if you have a sequence such as the following, it seems to me very confusing, especially for new users of R:


``` r
library(modelsummary); library(fixest)
mod <- feols(hp ~ mpg + drat | cyl, mtcars)
summary(mod, se = "standard")
#> OLS estimation, Dep. Var.: hp
#> Observations: 32 
#> Fixed-effects: cyl: 3
#> Standard-errors: Standard 
#>      Estimate Std. Error t value Pr(>|t|)    
#> mpg   -4.3939     2.0187 -2.1766 0.038425 *  
#> drat  39.8630    16.5210  2.4129 0.022889 *  
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#> RMSE: 31.4     Adj. R2: 0.75137 
#>              Within R2: 0.243173
msummary(mod, vcov = "iid", output = "markdown")
```

|             | Model 1  |
|:------------|:--------:|
| mpg         |  -4.394  |
|             | (1.697)  |
| drat        |  39.863  |
|             | (36.235) |
| Std. Errors |   IID    |
| FE: cyl     |    X     |

<sup>Created on 2021-05-07 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

I think it really takes some good experience with R to wrap your head around these results. And even for more experienced users, it seems possible to trip up here. 

I realize that this is not very constructive, as I don't have a good idea for how to solve this except to maintain a strict whitelist. One could also maintain a strict whitelist, but have two exceptions for `fixest` and `estimatr`, where the 'convenient' adjustments are _only_ allowed if the model was estimated without standard error adjustments--it should be possible to find out from inspection of the model objects. But of course, this means introducing package-specific code that has to be maintained.

Personally, I lean towards erroring more instead of reporting possibly wrong SEs, but you already know that :)  I would also be interested to hear about @grantmcdermott 's perspective on this.

**Comment by vincentarelbundock** [comment]
Oh, that's very interesting!

Curiously, the `sandwich::vcovHC` function returns a matrix of `NA`s:

``` r
library("fixest")
library("sandwich")

mod <- feols(hp ~ mpg + drat | cyl, mtcars, se = "cluster")

vcovHC(mod)
#> hatvalues.fixest: 'hatvalues' is not implemented for estimations with fixed-effects.
#>      mpg drat
#> mpg   NA   NA
#> drat  NA   NA

vcovCL(mod, ~cyl)
#>            mpg       drat
#> mpg    2.69460  -56.99115
#> drat -56.99115 1228.23843

vcovBS(mod)
#>            mpg     drat
#> mpg   1.886894  -8.9010
#> drat -8.901000 377.9967
```

In contrast, the `sandwich` functions simply don't seem to work on `estimatr` objects. This is probably better.

My current view (formed at 6:39AM, so TIFWIW), is that this should probably be fixed upstream. I'm not sure that `modelsummary` should be responsible for incorrect results obtained when calling `vcovCL` on an object that it did not create itself and whose features it has no way to anticipate.

In this particular case, maybe we can hardcode a warning to the user because we know there's a tricky incompatibility issue between `fixest` and `sandwich`, but again I wouldn't want to maintain a list of a dozen such instances.

If the results really are wrong when applying `sandwich` functions, maybe this is worth reporting on the `fixest` repository.

**Comment by grantmcdermott** [comment]
Hi guys,

I'm going to have to be brief, so just jotting down some quick thoughts:

1. I know that @lrberge is planning to introduce a "vcov" argument in the next version of `fixest`. This single argument will supersede the current, separate "se" and "cluster" arguments. I'm not entirely sure what impact this will have on the package's compatibility with `sandwhich`. But my understanding is that it partly motivated by improving consistency of user experience. So, some of this downstream trickiness may be resolved on its own. (Laurent, please correct me if I am wrong on any details.)
2. With that in mind, I agree that we should hold off on hard-coding any particular functionality in `modelsummary` for now; apart perhaps from the suggested warning about naively passing "vcov" arguments down from `modelsummary` to fixest objects.

**Comment by vincentarelbundock** [comment]
thanks @grantmcdermott this is useful

**Comment by elbersb** [comment]
I have to say, I'm not quite sure how all of these packages interact. My understanding is that currently `modelsummary` will readjust already-adjusted variance-covariance matrices that are coming from `fixest`, but I might be wrong. 

I also think it would be a good idea to have a warning when applying the "convenient" vcov arguments to `fixest` and `estimatr` objects. For instance, I feel that a warning would be appropriate here:

``` r
library(estimatr)
library(fixest)
library(modelsummary)

mod1 <- lm_robust(mpg ~ disp, data = mtcars)
mod2 <- feols(mpg ~ disp | cyl, mtcars)
modelsummary(list(mod1, mod2), vcov = "iid", output = "markdown")
```

|             | Model 1 | Model 2 |
|:------------|:-------:|:-------:|
| (Intercept) | 29.600  |         |
|             | (1.490) |         |
| disp        | -0.041  | -0.027  |
|             | (0.005) | (0.015) |
| Std. Errors |   IID   |   IID   |
| FE: cyl     |         |    X    |
| se\_type    |   HC2   |         |

<sup>Created on 2021-05-09 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

I'd prefer to be even stricter, though -- modelsummary claims to show iid standard errors, although it can't be sure that they are. In the long run, I think a rather strict whitelist for the "convenient" arguments would avoid ambiguities. If users want to adjust `fixest` standard errors, it's easy to do through fixest's `vcov` function, which can be passed to `modelsummary`.

**Comment by vincentarelbundock** [comment]
Yes, I understand that the status quo is not ideal. However, both the blacklist and whitelist approaches seem very costly to implement.

1. **Whitelist** approach requires a list of all models for which `vcov(model)` actually returns an IID vcov matrix.
2. **Blacklist** approach requires a list of all models for which `vcov(model)` does **not** return an IID vcov matrix.

This is not just about `sandwich`. Nearly all of the 100+ supported models use a `vcov` method, so both approaches would require us to collect information about the statistical features of the matrices produced by `vcov` for all of those models. Whether we want to raise an error or a warning is a discussion we can have, but that pre-supposes data collection on hundreds of models.

Honestly, I don't really have the energy for this. So, for now, I have just improved the documentation. But I'm still open to concrete ideas for solutions that would (a) still allow us to use these convenient shortcuts like "iid", (b) be stricter, and (c) does not require us to compile information on all the supported models.

**Comment by grantmcdermott** [comment]
I think the question is: How many model classes / packages offer a post-estimation VCOV adjustment method that is distinct from `sandwich`? Off the top of my head, I can only think of two: `fixest` and `lfe`. 

In the case of `lfe' it's not really a post-hoc adjustment, so much as multiple vcov matrices automatically being retained with the returned model object. Indeed, I wrote the broom::tidy method for it just to "emulate" similar behavior to sandwich, so I don't think it can cause trouble.

In that case, we're left with fixest and estimatr, as the potential trouble cases. The former may be resolved automatically with the native "vcov" argument. The latter, I think you simply have to blacklist.

Have I missed any others?

**Comment by vincentarelbundock** [comment]
Maybe `plm`, but aside from that I think you're right that the range of problematic models is limited.

So the blacklist approach is to raise an error (or warning) if `vcov="iid"` or `vcov="classical"` for a model of the known problematic classes?

Should we introspect the models before deciding to warn? (That seems like trouble, so probably not.)

Error or warning?

**Comment by grantmcdermott** [comment]
plm and ivreg, etc. all use sandwich IIRC.

I'm typing on my phone whilst out in the garden helping my son with planting. I promise to give this some more thought tonight, though!

**Comment by elbersb** [comment]
Not sure if more input from my part is at all useful here, but I'm preparing a workshop right now and keep running into these cases. So here's another one:

``` r
data("SchoolingReturns", package = "ivreg")
library("fixest")
library("modelsummary")

iv <- feols(log(wage) ~ ethnicity + experience + smsa | 
  education ~ nearcollege, data = SchoolingReturns)

msummary(iv, vcov = list(NULL, "robust", vcov(iv, se = "hetero")),
    gof_omit = "R2|IC|Log", output = "markdown")
```

|                | Model 1  | Model 2  | Model 3  |
|:---------------|:--------:|:--------:|:--------:|
| (Intercept)    |  3.121   |  3.121   |  3.121   |
|                | (0.781)  | (19.187) | (0.777)  |
| fit\_education |  0.178   |  0.178   |  0.178   |
|                | (0.047)  | (1.154)  | (0.047)  |
| ethnicityafam  |  -0.118  |  -0.118  |  -0.118  |
|                | (0.057)  | (1.320)  | (0.056)  |
| experience     |  0.080   |  0.080   |  0.080   |
|                | (0.019)  | (0.457)  | (0.019)  |
| smsayes        |  0.128   |  0.128   |  0.128   |
|                | (0.032)  | (0.671)  | (0.032)  |
| Num.Obs.       |   3010   |   3010   |   3010   |
| Std. Errors    | Standard | Standard | Standard |

<sup>Created on 2021-05-10 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

```r
> sqrt(diag(sandwich::vcovHC(iv)))
  (Intercept) fit_education ethnicityafam    experience       smsayes 
   19.1868140     1.1542289     1.3203443     0.4567060     0.6706925 
```

To me, this seems as if in this case `sandwich` is actually the problem, no? I'm not quite sure why `vcovHC` accepts a `fixest` object in the first place. I don't have much time today, but will try to look into it.

**Comment by vincentarelbundock** [comment]
I would not call this a "problem" with `sandwich`. The issue is that `fixest` is extremelly **well** design, and that it offers all the standard methods (`vcov`, `hatvalues`, etc.). The `vcovHC.default` method from `sandwich` is written in a very generic way to work with as many models as possible; it only uses the "standard" methods like `vcov`, `hatvalues`, etc.

Because both functions are generic and well designed, they worked together seamlessly, even if they were not explicitly intended to or designed in this way.

This is just a weird situation where everything works well, but we still get an unintended result.

I understand the problem but don't yet have a clear view of the best solution. I'll think about it some more and do something as soon as lightning strikes.

**Comment by lrberge** [comment]
Sorry, late to the party! I'm in the middle of a 1200km move (literally. Now I'm in a hotel room, the truck is outside, fully loaded, and before leaving early tomorrow, I'll have a night without worries nor kids: that's paradise for one night!).

I can't bring any thoughtful suggestion but I'd just add that `fixest` compatibility with `sandwich` is not great (problem with degrees of freedom for clustered SEs, `hatvalues` cannot be computed with fixed-effects (it requires stuff like X'X, not possible to have for large FEs), differences b/w an estimation with FEs and the same estimation with dummies (eg HAC selects a different bandwidth) and it looks like there is a specific issue with `fixest` IV estimations above that I've just discovered). But it also works in many situations... (the IV pblm is very large though, I'll try to do stg.)

@vincentarelbundock: I tried to look at the code but couldn't find how `iid` VCOVs are computed. It seems that [get_vcov](https://github.com/vincentarelbundock/modelsummary/blob/b1579f39ee91c050713c4d98986c0ee58f613e96/R/get_vcov.R#L25 ) returns NULL when it bumps into an iid VCOV, so I guess it is computed somewhere else. Is it a method that is applied? Or is it fetched from the object?
If I can do anything to help in `fixest` code, I'd be happy to, but regarding `iid` SEs I don't see how so far.

> If the results really are wrong when applying sandwich functions, maybe this is worth reporting on the fixest repository.

Agreed. And I also 100% agree that you shouldn't be the arbiter of `fixest`'s (in)compatibilities with `sandwich`, it's not your role and it's too much work (and it's not one off since compatibility is subject to change).

Btw @grantmcdermott is right: a new `vcov` argument will be there in the next version of `fixest`, and what he doesn't say is that he's convinced me to use `iid` instead of `standard` for the default standard-error (and this name is indeed much better! :-)).

**Comment by vincentarelbundock** [comment]
Wow, good luck with the move @lrberge! These things are always such an ordeal. At least I hope that you're heading toward fun adventures!

To clarify, the `vcov="iid"` shortcut in `modelsummary` is extremely dumb: when `get_vcov` returns `NULL`, then `modelsummary` automatically uses the default standard errors extracted by `broom::tidy` (in other words:  vanilla `vcov(model)`).

Allowing this dumb alias/shortcut is useful because `modelsummary` can then insert a nice label in the table, and it also makes sense because in the vast majority of cases it will be correct. However, I must admit that I did not anticipate that `vcov(model)` would return non-standard covariance matrices from time to time (here: when `feols` is called with the `se` argument.

This is a problem with `modelsummary`'s dumb "iid" alias, and not a problem with `fixest`.

I will probably just raise a warning or error for the handful of models that do not (always) return IID uncertainties when calling `vcov(model)`. Not a big deal, but it's getting a bit late for me to code right now ;)

**Comment by elbersb** [comment]
Caught the edge case below, not sure if it's worth fixing.

``` r
library("modelsummary")
library("fixest")
mod <- feols(mpg ~ cyl | am, data = mtcars)
modelsummary(mod, vcov = list("IID" = vcov(mod, se = "standard")), output = "markdown")
#> Warning in get_list_of_modelsummary_lists(models = models, conf_level =
#> conf_level, : When the `vcov` argument is set to "iid", "classical", or
#> "constant", `modelsummary` extracts the default variance-covariance matrix from
#> the model object. For objects of class `fixest`, the default vcov is not always
#> IID. Please make sure that the standard error label matches the numeric results
#> in the table. Note that the `vcov` argument accepts a named list for users who
#> want to customize the standard error labels in their regression tables.
```

|             | Model 1 |
|:------------|:-------:|
| cyl         | -2.501  |
|             | (0.361) |
| Num.Obs.    |   32    |
| R2          |  0.759  |
| R2 Adj.     |  0.742  |
| R2 Within   |  0.624  |
| R2 Pseudo   |         |
| AIC         |  165.2  |
| BIC         |  169.6  |
| Log.Lik.    | -79.610 |
| Std. Errors |   IID   |
| FE: am      |    X    |

<sup>Created on 2021-05-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by vincentarelbundock** [comment]
Thanks! 10f7ddf
